### PR TITLE
Make MemoryFile.openRead and _ChrootFile.openRead return Stream<List<int>> again

### PR DESCRIPTION
**Pull request by jamesderlin**
This is equivalent to https://github.com/google/file.dart/pull/168, but I foolishly neglected to make corresponding changes for all file system types.

Fixes https://github.com/google/file.dart/issues/193.

### CODE REVIEW

**Comment by jamesderlin** [comment]
This would be a breaking change so probably should wait for the next major version change (perhaps https://github.com/google/file.dart/pull/210).

**Comment by dnfield** [comment]
Why does glob depend on file?

**Comment by dnfield** [comment]
@jakemac53 WDYT? Looks like you introduced the file dep on glob here: https://github.com/dart-lang/glob/commit/ba29990778bb2f8ad884cd4e5ab654cb766e3e87

**Comment by jakemac53** [comment]
> And based on the CI failures, bumping the major version apparently is a can of worms because `file` depends on `test` which depends on `test_core` which depends on `glob` which depends on `file`... How was this handled last time?

I believe you could fix this by overriding the dependency on `glob`:

```
dependency_overrides:
  glob: 2.1.1
```

(or if that doesn't fix it, some other combination of overrides will).

> @jakemac53 WDYT? Looks like you introduced the file dep on glob here: [dart-lang/glob@ba29990](https://github.com/dart-lang/glob/commit/ba29990778bb2f8ad884cd4e5ab654cb766e3e87)

Glob depends on file because it supports globbing file systems - and we generalized that support to arbitrary file systems using the `package:file` apis, which allowed us to drop the required `dart:io` dependency, and makes glob compatible with any platform.

**Comment by jakemac53** [comment]
(note that the dependency override is only for a dev dependency so it isn't an issue, and once we publish a new package:glob that allows 7.x then it can be removed)

**Comment by jakemac53** [comment]
Note that as a part of `7.x` we should also do https://github.com/google/file.dart/issues/209

**Comment by jakemac53** [comment]
The other breaking change (migrate to Dart 3) has landed now.

**Comment by jamesderlin** [comment]
Hm, GitHub seems to be confused by me rebasing, resolving the conflicts, and force-pushing.  I suppose I'll have to recreate the PR.

**Comment by bernaferrari** [comment]
What's the issue?

**Comment by jamesderlin** [comment]
GitHub didn't pick up my updated changes from force-pushing, but it looks like it managed to sort itself out eventually, so I don't need to recreate the PR after all.

This PR still needs a review though.

**Review by jakemac53 — APPROVED** [review]
cc @natebosch could you also take a look?

File: packages/file/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 * Dart 3 fixes for class modifiers.
 
+* Make the return type of `MemoryFile.openRead` and `_ChrootFile.openRead` again
+  match the return type from `dart:io`.
```
Comment (natebosch) [inline packages/file/CHANGELOG.md:6]: [nit]

```suggestion
* Make the return type of `MemoryFile.openRead` and `_ChrootFile.openRead` again
  match the return type from `dart:io`.
```

File: packages/file/lib/src/backends/memory/memory_file.dart [hunk 2]
```diff
@@ -206,9 +206,9 @@ class MemoryFile extends MemoryFileSystemEntity implements File {
             ? content.sublist(start)
             : content.sublist(start, math.min(end, content.length));
       }
-      return Stream<Uint8List>.fromIterable(<Uint8List>[content]);
+      return Stream<List<int>>.fromIterable(<Uint8List>[content]);
```
Comment (natebosch) [inline packages/file/lib/src/backends/memory/memory_file.dart:209]:
```suggestion
      return Stream.value(content);
```

File: packages/file/lib/src/backends/memory/memory_file.dart [hunk 3]
```diff
@@ -206,9 +206,9 @@ class MemoryFile extends MemoryFileSystemEntity implements File {
...
     } catch (e) {
-      return Stream<Uint8List>.fromFuture(Future<Uint8List>.error(e));
+      return Stream<List<int>>.fromFuture(Future<Uint8List>.error(e));
```
Comment (natebosch) [inline packages/file/lib/src/backends/memory/memory_file.dart:211]:
```suggestion
      return Stream.error(e);
```

**Review by natebosch — APPROVED** [review]

**Comment by jamesderlin** [comment]
Thanks for the reviews! (@jakemac53 or @natebosch, could I trouble either of you to take a look at the other `package:file` PRs too?)

**Pull request merged**